client/extensions/woocommerce/state/selectors/promotions.js

```diff
@@ -1,31 +1,60 @@
 /**
  * External dependencies
  */
-import { get } from 'lodash';
+import { get, isObject } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import { getSelectedSiteWithFallback } from '../sites/selectors';
 
 export function getPromotions( rootState, siteId = getSelectedSiteWithFallback( rootState ) ) {
-	const { promotions } = get( rootState, [ 'extensions', 'woocommerce', 'sites', siteId ], {} );
+	const promotions = get( rootState, [ 'extensions', 'woocommerce', 'sites', siteId, 'promotions' ], {} );
 	return promotions.promotions;
 }
 
+export function getPromotion( rootState, promotionId, siteId = getSelectedSiteWithFallback( rootState ) ) {
+	const promotions = getPromotions( rootState, siteId );
+	return promotions.find( ( p ) => promotionId === p.id ) || null;
+}
+
 export function getPromotionsPage( rootState, siteId = getSelectedSiteWithFallback( rootState ), page, perPage ) {
 	const offset = ( page - 1 ) * perPage;
 	const promotions = getPromotions( rootState, siteId );
 	return ( promotions ? promotions.slice( offset, ( offset + perPage ) ) : null );
 }
 
 export function getPromotionsCurrentPage( rootState ) {
-	const { list } = get( rootState, [ 'extensions', 'woocommerce', 'ui', 'promotions' ], {} );
+	const list = get( rootState, [ 'extensions', 'woocommerce', 'ui', 'promotions', 'list' ], {} );
 	return list.currentPage;
 }
 
 export function getPromotionsPerPage( rootState ) {
-	const { list } = get( rootState, [ 'extensions', 'woocommerce', 'ui', 'promotions' ], {} );
+	const list = get( rootState, [ 'extensions', 'woocommerce', 'ui', 'promotions', 'list' ], {} );
 	return list.perPage;
 }
 
+export function getCurrentlyEditingPromotionId( rootState, siteId = getSelectedSiteWithFallback( rootState ) ) {
+	const edits = get( rootState, [ 'extensions', 'woocommerce', 'ui', 'promotions', 'edits', siteId ], {} );
+	return edits.currentlyEditingId || null;
+}
+
+export function getPromotionEdits( rootState, promotionId, siteId = getSelectedSiteWithFallback( rootState ) ) {
+	const edits = get( rootState, [ 'extensions', 'woocommerce', 'ui', 'promotions', 'edits', siteId ], {} );
+
+	if ( isObject( promotionId ) ) {
+		return ( edits.creates && edits.creates.find( ( p ) => promotionId === p.id ) ) || null;
```

Comment:
if `promotionId` is an object, will/can this clause ever be true? looking at the test cases, it seems like `promotionId` is only ever tested as a string value - do we need other test cases here or should this just be `if promotionId`?

Reply:
Good advice. I added a test for this. Thanks.